packages/host/app/components/matrix/room.gts

```diff
@@ -534,10 +534,19 @@ export default class Room extends Component<RoomArgs> {
   });
 
   private doSetObjective = restartableTask(async () => {
+    // objective are currently non-primitive fields
     let catalogEntry = await chooseCard<CatalogEntry>({
       filter: {
-        on: catalogEntryRef,
-        eq: { isPrimitive: false },
+        every: [
+          {
+            on: catalogEntryRef,
+            eq: { isField: true },
+          },
+          {
+            on: catalogEntryRef,
+            eq: { isPrimitive: false },
```

Comment:
do we need both of these? Can you have an entry that is a primitive but not a field??

Reply:
it's more like the other way around, do you want to be able to find catalog entries that are fields but not primitives. The answer is yes--room objectives currently fit this criteria. when you search for a room objective to add it is a non-primitive field.